storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1374,5 +1393,29 @@ public static RuntimeException wrapInRuntime(Exception e){
             return new RuntimeException(e);
         }
     }
+
+    public static int getAvailablePort(Integer port) {
+        Integer localPort = null;
+        ServerSocket socket = null;
+        try {
+            socket = new ServerSocket(port);
+            localPort = socket.getLocalPort();
+        } catch (IOException exp) {
+            getAvailablePort(0);
+        } finally {
```

Comment:
I just thought when we are converting the tests to java auto boxing will help to remove the specific conversion, I just wanted to return a null rather than just returning. Moreover, i might have to wrap all calls for clojure in int as the default is long.